### PR TITLE
fix(thinking): Add support for claude thinking and make chat more robust

### PR DESCRIPTION
**Pull request by zereraz**
### Description
Claude thinking required special arguments to be sent to bedrock.
Improve chat streams, ensure it works well for reasoning and non reasoning. Manage various edge cases of invalid json generation by llms.

### Testing
Added more test cases and tested locally.

## Summary by CodeRabbit

- **New Features**
	- Enhanced AI assistant to provide answers in well-formatted markdown within the response.
	- Introduced specialized reasoning mode for certain AI models, improving transparency in the assistant’s thought process.
- **Improvements**
	- Increased maximum token limit for AI responses, enabling longer outputs.
	- Improved handling and parsing of AI-generated JSON responses, resulting in more robust and reliable outputs.
	- Enhanced citation formatting and error handling in AI responses.
	- Configured retry behavior for AI service client to improve reliability.
	- Refined logic for extracting reasoning tokens from streamed AI responses.
	- Adjusted email search filters to better target relevant messages.
- **Bug Fixes**
	- Fixed issues with extracting and displaying reasoning content from streaming AI responses.
- **Tests**
	- Expanded test coverage for JSON parsing to ensure correct handling of various edge cases.

### CODE REVIEW

File: server/ai/provider/bedrock.ts [hunk 1]
```diff
@@ -68,54 +68,88 @@ export class BedrockProvider extends BaseProvider {
...
+    const inferenceConfig = isThinkingEnabled
+      ? {
+          maxTokens: modelParams.maxTokens || 2500,
+          temperature: temperature,
+        }
+      : {
+          maxTokens: modelParams.maxTokens || 2500,
+          topP: modelParams.topP || 0.9,
+          temperature: temperature,
+        }
```
[the remark this answers is not in the document]
Comment (zereraz) [reply server/ai/provider/bedrock.ts:100]: will be done later.

File: server/ai/provider/index.ts [hunk 2]
```diff
@@ -417,7 +443,7 @@ export const jsonParseLLMOutput = (text: string): any => {
         .replace(/\r/g, "\\r")
         .trim()
       if (!text) {
-        return ""
+        return {}
```
[the remark this answers is not in the document]
Comment (zereraz) [reply server/ai/provider/index.ts:446]: the issue is if we return null and then the callee runs `parsed.answer` but we just set parsed to null
